### DISCUSSION TITLE
Edit the data in execute function and still saves the checksum correct

### DISCUSSION THREAD

**Discussion by itaimondshine** [ickerwx/tcpproxy#44] [General]
Hey.
I am using this tool for manipulating modbus packets. 
I am getting the tcp modbus as input to the execute function and return the data with additional string.
Is there a way to fix the checksum to be the correct one?

Thank you in advance, 
itai

**Comment by ickerwx** [comment]
Hi.

which checksum are you talking about? If you are referring to the CRC used in Modbus, then you will have to recalculate it yourself. You'll probably have to parse the modbus messages for this to reliably find the position of the checksum field. It's been quite a while since I worked with Modbus, I'm not much help here, sorry.
